src/main/native/compression/IntelDeflater.cc

```diff
@@ -258,9 +296,32 @@ JNIEXPORT jint JNICALL Java_com_intel_gkl_compression_IntelDeflater_deflateNativ
       
     z_stream* lz_stream = (z_stream*)env->GetLongField(obj, FID_lz_stream);
 
+    if (lz_stream == NULL) {
```

Comment:
This goes for the rest of these. 

We need a Macro or a fuction to handle this.  There is alot of code duplication.  It looks like just a few wrapper functions macros would do. 

something like: 

glk_GetLogField(obj, type,message)

Reply:
As we discussed, I'll leave it as is.